core/src/main/java/com/datastrato/gravitino/connector/capability/HasCapabilities.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.connector.capability;
+
+import com.datastrato.gravitino.annotation.Evolving;
+
+/**
+ * The Catalog interface to provide the capabilities of the catalog. If the implemented catalog has
+ * some special capabilities, it should override the default implementation of the capabilities.
+ */
+@Evolving
+public interface HasCapabilities {
```

Comment:
I think the name `HasCapabilities` is not so proper here. I would call it `Capability`.

If you want to define a `HasCapabilities` interface? It should be like:

```java

interface HasCapabilities {
  Capability[] hasCapabilies()
}
```

Reply:
updated